File: src/main/java/com/clinica/atendimento/model/converters/GrauParentescoConverter.java
package com.clinica.atendimento.model.converters;

import com.clinica.atendimento.model.enums.GrauParentesco;
import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

@Converter
public class GrauParentescoConverter implements AttributeConverter<GrauParentesco, String> {

    @Override
    public String convertToDatabaseColumn(GrauParentesco attribute) {
        if (attribute == null) {
            return null;
        }
        return attribute.getCodigo();
    }

    @Override
    public GrauParentesco convertToEntityAttribute(String dbData) {
        if (dbData == null) {
            return null;
        }
        return GrauParentesco.fromCodigo(dbData);
    }
}

File: src/main/kotlin/com/clinica/atendimento/model/converters/GrauParentescoConverter.kt
package com.clinica.atendimento.model.converters

import com.clinica.atendimento.model.enums.GrauParentesco
import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter
class GrauParentescoConverter : AttributeConverter<GrauParentesco, String> {

    override fun convertToDatabaseColumn(attribute: GrauParentesco?): String? {
        return attribute?.codigo
    }

    override fun convertToEntityAttribute(dbData: String?): GrauParentesco? {
        return if (dbData == null) null else GrauParentesco.fromCodigo(dbData)
    }
}